### PR TITLE
Use widget ID to style each widget; CSS to footer

### PR DESCRIPTION
**Pull request by damiencarbery**
Record instance ID of each active widget.
Move CSS to footer and use the recorded IDs to generate the correct CSS for each active widget.
Suggested fix for #8

### CODE REVIEW

**Comment by carasmo** [comment]
Yay! I'm going to install this PR.

**Comment by carasmo** [comment]
Thank you!

**Comment by marksabbath** [comment]
Made the changes and merged. Thanks for contributing @damiencarbery really appreciate!

File: simple-social-icons.php [hunk 1]
```diff
@@ -425,38 +436,44 @@ function css() {
...
-			border-radius: ' . $instance['border_radius'] . 'px;
-			color: ' . $instance['icon_color'] . ' !important;
-			border: ' . $instance['border_width'] . 'px ' . $instance['border_color'] . ' solid !important;
-			font-size: ' . $font_size . 'px;
-			padding: ' . $icon_padding . 'px;
-		}
+		$css = '';
+		foreach ( $this->active_instances as $instance_id ) {
+			// Skip if info for this instance does not exist - this should never happen.
+			if (! isset( $all_instances[ $instance_id ] ) ) {
```
[the remark this answers is not in the document]
Comment (damiencarbery) [reply simple-social-icons.php]: I’m on vacation at the moment and not back until Monday. You can make those WordPress Coding Standards changes if you want them done before then.

**Comment by JiveDig** [comment]
Just tested this with multiple instances on the same page (3) and it worked flawlessly. This is a much needed update to keep SSI relevant and at the top of the list for doing what it does ;)  Nice work!

**Pull request merged**